Update visibility explanations (#29263)
sourcegraph/sourcegraph: doc/code_insights/explanations/viewing_code_insights.md

There are three possible visibility levels:

- Private: visible only to you
- Shared with [an organization](../../../admin/organizations.md): visible to everyone in the organization
- Global: visible to everyone on the Sourcegraph instance

### Built-in dashboards

A default "dashboard" of all insights visible to a user appears as the homepage for the Insights navigation bar item. 

To add insights to your own custom dashboard, see [Creating a custom dashboard of code insights](../how-tos/creating_a_custom_dashboard_of_code_insights.md).

### Dashboard visibility

> A Dashboard's visibility level owns the visibility of code insights on the dashboard. If you add an Insight that was on a private dashboard to an organization or global dashboard, now people with access to that dashboard can view the insight. When you first create an insight, it defaults to appearing on the dashboard from which you clicked the "create" button. TODO::CONFIRM If you create an insight directly on the create page or from the default dashboard, it [seems to appear on the default dashboard? but what is the visibility?]. 

Reviewer: Confirmed this is correct. Source: tried it locally and that's what happened 😅

> A Dashboard's visibility level owns the visibility of code insights on the dashboard. If you add an Insight that was on a private dashboard to an organization or global dashboard, now people with access to that dashboard can view the insight. When you first create an insight, it defaults to appearing on the dashboard from which you clicked the "create" button. TODO::CONFIRM If you create an insight directly on the create page or from the default dashboard, it [seems to appear on the default dashboard? but what is the visibility?]. 

Author: ```suggestion
A Dashboard's visibility level owns the visibility of code insights on the dashboard. If you add an Insight that was on a private dashboard to an organization or global dashboard, now people with access to that dashboard can view the insight. When you first create an insight, it defaults to appearing on the dashboard from which you clicked the "create" button. If you create an insight directly on the create page or from the default dashboard, it will appear on your default dashboard. 
```
